<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Alex Sanchez">
<meta name="dcterms.date" content="2025-04-13">

<title>Rean√°lisis de resultados procesados - Estudio ST002993</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/libs/clipboard/clipboard.min.js"></script>
<script src="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/libs/quarto-html/quarto.js"></script>
<script src="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/libs/quarto-html/popper.min.js"></script>
<script src="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/libs/quarto-html/anchor.min.js"></script>
<link href="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#informaci√≥n-del-estudio" id="toc-informaci√≥n-del-estudio" class="nav-link active" data-scroll-target="#informaci√≥n-del-estudio">Informaci√≥n del estudio</a></li>
  <li><a href="#cargar-archivos" id="toc-cargar-archivos" class="nav-link" data-scroll-target="#cargar-archivos">Cargar archivos</a>
  <ul class="collapse">
  <li><a href="#resultados" id="toc-resultados" class="nav-link" data-scroll-target="#resultados">Resultados</a></li>
  </ul></li>
  <li><a href="#inspecci√≥n-b√°sica-y-combinaci√≥n" id="toc-inspecci√≥n-b√°sica-y-combinaci√≥n" class="nav-link" data-scroll-target="#inspecci√≥n-b√°sica-y-combinaci√≥n">Inspecci√≥n b√°sica y combinaci√≥n</a>
  <ul class="collapse">
  <li><a href="#metadatos" id="toc-metadatos" class="nav-link" data-scroll-target="#metadatos">Metadatos</a></li>
  <li><a href="#combinar-datos-y-metadatos" id="toc-combinar-datos-y-metadatos" class="nav-link" data-scroll-target="#combinar-datos-y-metadatos">Combinar datos y metadatos</a>
  <ul class="collapse">
  <li><a href="#acceso-a-los-slots" id="toc-acceso-a-los-slots" class="nav-link" data-scroll-target="#acceso-a-los-slots">Acceso a los ‚Äúslots‚Äù</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#preprocesamiento" id="toc-preprocesamiento" class="nav-link" data-scroll-target="#preprocesamiento">Preprocesamiento</a>
  <ul class="collapse">
  <li><a href="#filtrado" id="toc-filtrado" class="nav-link" data-scroll-target="#filtrado">Filtrado</a></li>
  <li><a href="#normalizaci√≥n" id="toc-normalizaci√≥n" class="nav-link" data-scroll-target="#normalizaci√≥n">Normalizaci√≥n</a></li>
  </ul></li>
  <li><a href="#exploraci√≥n-de-los-datos" id="toc-exploraci√≥n-de-los-datos" class="nav-link" data-scroll-target="#exploraci√≥n-de-los-datos">Exploraci√≥n de los datos</a>
  <ul class="collapse">
  <li><a href="#an√°lisis-de-correlacion" id="toc-an√°lisis-de-correlacion" class="nav-link" data-scroll-target="#an√°lisis-de-correlacion">An√°lisis de correlacion</a></li>
  <li><a href="#analisis-factoriales" id="toc-analisis-factoriales" class="nav-link" data-scroll-target="#analisis-factoriales">Analisis factoriales</a></li>
  <li><a href="#mfa" id="toc-mfa" class="nav-link" data-scroll-target="#mfa">MFA</a>
  <ul class="collapse">
  <li><a href="#matriz-de-factores" id="toc-matriz-de-factores" class="nav-link" data-scroll-target="#matriz-de-factores">Matriz de factores</a></li>
  </ul></li>
  <li><a href="#clustering-y-obesidad" id="toc-clustering-y-obesidad" class="nav-link" data-scroll-target="#clustering-y-obesidad">Clustering y obesidad</a>
  <ul class="collapse">
  <li><a href="#clustering-y-relaci√≥n-con-la-obesidad" id="toc-clustering-y-relaci√≥n-con-la-obesidad" class="nav-link" data-scroll-target="#clustering-y-relaci√≥n-con-la-obesidad">üîç Clustering y relaci√≥n con la obesidad</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#almacenaje-de-datos-y-resultados" id="toc-almacenaje-de-datos-y-resultados" class="nav-link" data-scroll-target="#almacenaje-de-datos-y-resultados">Almacenaje de datos y resultados</a></li>
  </ul>
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="Rean√°lisis_de_resultados_procesados-Estudio_ST002993.pdf"><i class="bi bi-file-pdf"></i>PDF</a></li></ul></div></nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Rean√°lisis de resultados procesados - Estudio ST002993</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Alex Sanchez </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">April 13, 2025</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="informaci√≥n-del-estudio" class="level1">
<h1>Informaci√≥n del estudio</h1>
<p>Este estudio se basa en el proyecto depositado en MetabolomicsWorkbench:</p>
<p><a href="">https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Project&amp;ProjectID=PR001863</a></p>
<p>EL resumen del proyecto en el repositorio es el siguiente:</p>
<p><em>La obesidad es el resultado de una interacci√≥n entre la predisposici√≥n gen√©tica y factores ambientales como la dieta, la actividad f√≠sica, la cultura y el nivel socioecon√≥mico. Los tratamientos personalizados ser√≠an ideales, por lo que es necesario identificar caracter√≠sticas individuales que permitan mejorar la eficacia de las terapias. Por ejemplo, alteraciones gen√©ticas en la v√≠a leptina-melanocortina pueden provocar casos raros de obesidad grave de inicio temprano.</em></p>
<p><em>La metabol√≥mica tiene el potencial de distinguir entre un estado saludable y uno obeso; sin embargo, diferenciar subgrupos de individuos dentro del espectro de la obesidad sigue siendo un reto. El an√°lisis factorial puede integrar caracter√≠sticas cl√≠nicas y moleculares de los pacientes a partir de diversas fuentes, permitiendo una subclasificaci√≥n precisa.</em></p>
<p><em>Este estudio presenta un flujo de trabajo para identificar <strong>metabotipos</strong>, especialmente en contextos donde los estudios cl√≠nicos convencionales no logran categorizar adecuadamente a los pacientes. Se estudiaron 110 ni√±os con obesidad (IMC &gt; +2 SDS), genotipados para nueve genes implicados en la v√≠a leptina-melanocortina (CPE, MC3R, MC4R, MRAP2, NCOA1, PCSK1, POMC, SH2B1 y SIM1) y dos genes de receptores de glutamato (GRM7 y GRIK1). De ellos, 55 presentaban variantes raras heterocigotas y 55 no ten√≠an variantes.</em></p>
<p><em>Se recogieron datos antropom√©tricos y de laboratorio cl√≠nico rutinario, y las muestras de suero se procesaron para an√°lisis metabol√≥micos no dirigidos utilizando GC-q-MS, CE-TOF-MS y U(H)PLC-QTOF-MS/MS en modos de ionizaci√≥n positiva y negativa. Tras el procesamiento de se√±ales y la multialineaci√≥n, se aplicaron an√°lisis estad√≠sticos multivariantes y univariantes para evaluar la asociaci√≥n entre los rasgos gen√©ticos y los datos metabol√≥micos y cl√≠nicos.</em></p>
<p><em>Ni la presencia de variantes gen√©ticas raras heterocigotas ni los datos cl√≠nicos rutinarios permitieron identificar subgrupos dentro de los datos metabol√≥micos. Para descubrir subtipos metabol√≥micos, se aplic√≥ <strong>an√°lisis factorial<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a></strong> sobre una matriz compuesta construida a partir de las cinco plataformas<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a> anal√≠ticas. Se identificaron seis factores y tres metabotipos distintos. Se observaron diferencias sutiles pero claras en los l√≠pidos circulantes y en la sensibilidad a la insulina, lo que abre la posibilidad de personalizar el tratamiento en funci√≥n de la clasificaci√≥n del paciente dentro de estos subtipos de obesidad.</em></p>
<p><em>La identificaci√≥n de metabotipos en contextos cl√≠nicos presenta desaf√≠os debido a la influencia de m√∫ltiples variables no controladas sobre los fenotipos metab√≥licos. Sin embargo, esta estrategia demuestra su potencial para identificar subconjuntos de pacientes con diagn√≥sticos cl√≠nicos similares pero con condiciones metab√≥licas distintas. Este enfoque pone de relieve la aplicabilidad del an√°lisis factorial para el estudio de metabotipos en diversos escenarios cl√≠nicos.</em></p>
</section>
<section id="cargar-archivos" class="level1">
<h1>Cargar archivos</h1>
<p>Los datos se han descargado del repositorio a un subdirectorio ST002993. Para cada plataforma se dispone de dos tipos de archivo</p>
<ul>
<li>ST002993_AN0049XX.txt: Los metadatos</li>
<li>ST002993_AN0049XX_Results.txt: Los datos preprocesados</li>
</ul>
<p>XX: 13, 14, 16, 17</p>
<section id="resultados" class="level2">
<h2 class="anchored" data-anchor-id="resultados">Resultados</h2>
<p>En el estudio se usaron varias plataformas anal√≠ticas:</p>
<ul>
<li>AN004913 : LC-MS positivo</li>
<li>AN004914 ‚Üí LC-MS negativo</li>
<li>AN004916 ‚Üí CE-MS</li>
<li>AN004917 ‚Üí GC-MS</li>
</ul>
<p>Segn la tecnolog√≠a, la primera columna se indica como <code>Mass_RT</code> y <code>Mass_MT</code> que identifican cada pico metabol√≥mico combinando la masa (m/z) con el tiempo.</p>
<p><code>Mass_RT</code> se usa en LC-MS y GC-MS (tiempo de retenci√≥n), mientras que <code>Mass_MT</code> se usa en CE-MS (tiempo de migraci√≥n).</p>
<p>Ambos permiten identificar de forma √∫nica cada feature y pueden unificarse como <code>Feature_ID</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Cargar tidyverse para manejo de datos</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Directorio local donde has descargado los .txt</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="st">"../data"</span>  <span class="co"># CAMBIA esta ruta si es necesario</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>results_dir <span class="ot">&lt;-</span> <span class="st">"../results"</span> </span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Cargar cada archivo</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>res13 <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="fu">file.path</span>(data_dir, <span class="st">"ST002993_AN004913_Results.txt"</span>), <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>res14 <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="fu">file.path</span>(data_dir, <span class="st">"ST002993_AN004914_Results.txt"</span>), <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>res16 <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="fu">file.path</span>(data_dir, <span class="st">"ST002993_AN004916_Results.txt"</span>), <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>res17 <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="fu">file.path</span>(data_dir, <span class="st">"ST002993_AN004917_Results.txt"</span>), <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Renombrar la primera columna en todos</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(res13)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"Feature_ID"</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(res14)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"Feature_ID"</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(res16)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"Feature_ID"</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(res17)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"Feature_ID"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="inspecci√≥n-b√°sica-y-combinaci√≥n" class="level1">
<h1>Inspecci√≥n b√°sica y combinaci√≥n</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># A√±adir columna 'platform' antes de combinar, para mantener la fuente de cada feature</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>res13<span class="sc">$</span>platform <span class="ot">&lt;-</span> <span class="st">"LC_MS_pos"</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>res14<span class="sc">$</span>platform <span class="ot">&lt;-</span> <span class="st">"LC_MS_neg"</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>res16<span class="sc">$</span>platform <span class="ot">&lt;-</span> <span class="st">"CE_MS"</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>res17<span class="sc">$</span>platform <span class="ot">&lt;-</span> <span class="st">"GC_MS"</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Reordenar columnas para asegurar consistencia (Feature_ID primero)</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>reorder_cols <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  df[, <span class="fu">c</span>(<span class="st">"Feature_ID"</span>, <span class="fu">sort</span>(<span class="fu">setdiff</span>(<span class="fu">colnames</span>(df), <span class="fu">c</span>(<span class="st">"Feature_ID"</span>, <span class="st">"platform"</span>))), <span class="st">"platform"</span>)]</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>res13_ord <span class="ot">&lt;-</span> <span class="fu">reorder_cols</span>(res13)</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>res14_ord <span class="ot">&lt;-</span> <span class="fu">reorder_cols</span>(res14)</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>res16_ord <span class="ot">&lt;-</span> <span class="fu">reorder_cols</span>(res16)</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>res17_ord <span class="ot">&lt;-</span> <span class="fu">reorder_cols</span>(res17)</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Combinar y guardar</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>combined_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(res13_ord, res14_ord, res16_ord, res17_ord)</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(combined_df,<span class="fu">file.path</span>(results_dir, <span class="st">"combined_results_aligned.csv"</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Guardar plataforma y ID antes del preprocesamiento</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>feature_ids <span class="ot">&lt;-</span> combined_df<span class="sc">$</span>Feature_ID</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>platforms <span class="ot">&lt;-</span> combined_df<span class="sc">$</span>platform</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>data_matrix <span class="ot">&lt;-</span> combined_df[, <span class="sc">!</span>(<span class="fu">colnames</span>(combined_df) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Feature_ID"</span>, <span class="st">"platform"</span>))]</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(data_matrix) <span class="ot">&lt;-</span> feature_ids</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="metadatos" class="level2">
<h2 class="anchored" data-anchor-id="metadatos">Metadatos</h2>
<p>Los archivos de metadatos contienen la informaci√≥n de forma no estructurada por lo que se utilizan expresiones regulares para extraerla.</p>
<p>En primer lugar se obtiene el tipo de obesidad de cada individuo, determinado a partir de su an√°lisis gen√©tico.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>meta_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(data_dir, <span class="at">pattern =</span> <span class="st">"ST002993_AN0049[^_]*</span><span class="sc">\\</span><span class="st">.txt$"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>all_factors <span class="ot">&lt;-</span> <span class="fu">map_dfr</span>(meta_files, <span class="cf">function</span>(file) {</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  lines <span class="ot">&lt;-</span> <span class="fu">readLines</span>(file)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  factor_lines <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"SUBJECT_SAMPLE_FACTORS"</span>, lines, <span class="at">value =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  split_lines <span class="ot">&lt;-</span> <span class="fu">str_split</span>(factor_lines, <span class="st">"</span><span class="sc">\\</span><span class="st">t+"</span>, <span class="at">simplify =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>(split_lines[, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="at">sample_id =</span> V3, <span class="at">factor_info =</span> V4) <span class="sc">%&gt;%</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">obesity_type =</span> <span class="fu">str_remove</span>(factor_info, <span class="st">"Factor:"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(sample_id, obesity_type)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Eliminar duplicados (por si acaso)</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>all_factors <span class="ot">&lt;-</span> <span class="fu">distinct</span>(all_factors)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Verificar</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(all_factors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 √ó 2
  sample_id obesity_type        
  &lt;chr&gt;     &lt;chr&gt;               
1 ""        ""                  
2 "C842"    "Idiopathic obesity"
3 "C1306"   "Idiopathic obesity"
4 "C1317"   "Idiopathic obesity"
5 "C1331"   "Idiopathic obesity"
6 "C1336"   "Idiopathic obesity"</code></pre>
</div>
</div>
<p>Las filas de la tabla se reordenan para que tengan el mismo orden que las columnas d ela matriz de datos</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extraer el orden de las muestras (columnas)</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>sample_order <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data_matrix)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Limpiar nombres (por si acaso)</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>all_factors <span class="ot">&lt;-</span> all_factors <span class="sc">%&gt;%</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample_id =</span> <span class="fu">trimws</span>(sample_id))</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Paso 3: Filtrar y reordenar los metadatos seg√∫n los datos</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>metadata_aligned <span class="ot">&lt;-</span> all_factors <span class="sc">%&gt;%</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sample_id <span class="sc">%in%</span> sample_order) <span class="sc">%&gt;%</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">factor</span>(sample_id, <span class="at">levels =</span> sample_order))</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Paso 4: Verificaci√≥n</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(metadata_aligned<span class="sc">$</span>sample_id <span class="sc">==</span> sample_order)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 110</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Guardar</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(all_factors,<span class="fu">file.path</span>(results_dir, <span class="st">"ST002993_sample_metadata.csv"</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="combinar-datos-y-metadatos" class="level2">
<h2 class="anchored" data-anchor-id="combinar-datos-y-metadatos">Combinar datos y metadatos</h2>
<p>Disponemos de dos tipos de metadatos - El tipo de obesidad, metadato de las muestras - El tipo de plataforma, metadato de los metabolitos (se omite el hecho de que algunos podrian ser el mismo)</p>
<p>Aunque es posible guardarlos como una matyriz y dos vectores de factores, es m√°s elegante y pr√°ctico agruparlos en un contenedor de Bioconductor</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Crear rowData con informaci√≥n de plataforma</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>row_data <span class="ot">&lt;-</span> <span class="fu">DataFrame</span>(<span class="at">platform =</span> platforms, <span class="at">row.names =</span> feature_ids)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Crear colData con metadatos alineados</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>col_data <span class="ot">&lt;-</span> metadata_aligned <span class="sc">%&gt;%</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"sample_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">DataFrame</span>()</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Comprobar alineaci√≥n</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(<span class="fu">colnames</span>(data_matrix) <span class="sc">==</span> <span class="fu">rownames</span>(col_data)))</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(<span class="fu">rownames</span>(data_matrix) <span class="sc">==</span> <span class="fu">rownames</span>(row_data)))</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Crear objeto SummarizedExperiment con rowData</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">SummarizedExperiment</span>(</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">assays =</span> <span class="fu">list</span>(<span class="at">counts =</span> <span class="fu">as.matrix</span>(data_matrix)),</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">colData =</span> col_data,</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">rowData =</span> row_data</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Ver estructura</span></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>se</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: SummarizedExperiment 
dim: 892 110 
metadata(0):
assays(1): counts
rownames(892): 103.0999_0.77 103.9848_0.77 ... 125.0152_21.53
  367.1848_21.52
rowData names(1): platform
colnames(110): C1030 C1131 ... M962 M996
colData names(1): obesity_type</code></pre>
</div>
</div>
<section id="acceso-a-los-slots" class="level3">
<h3 class="anchored" data-anchor-id="acceso-a-los-slots">Acceso a los ‚Äúslots‚Äù</h3>
<p>Una vez creado el objeto <code>SummarizedExperiment</code> podemos extraer la matriz de expresi√≥n y los metadatos facilmente.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>matriz <span class="ot">&lt;-</span> <span class="fu">assay</span>(se)  <span class="co"># tambi√©n puedes usar assay(se, "counts") si hay varios assays</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co"># üîπ 2. Extraer los metadatos (informaci√≥n de las muestras)</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>infoMuestras <span class="ot">&lt;-</span> <span class="fu">colData</span>(se)</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>df_infoMuestras <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(infoMuestras)  <span class="co"># convertir a data.frame cl√°sico</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>infoVars <span class="ot">&lt;-</span> <span class="fu">rowData</span>(se)</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>df_infoVars <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(infoVars)  <span class="co"># convertir a data.frame cl√°sico</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co"># üîπ 3. Ver nombres de muestras y metabolitos</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>muestras <span class="ot">&lt;-</span> <span class="fu">colnames</span>(se)</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>metabolitos <span class="ot">&lt;-</span> <span class="fu">rownames</span>(se)</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Mostrar ejemplos</span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(matriz[, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                 C1030    C1131     C1133     C1171     C1204
103.0999_0.77 38821.76 40447.97  56677.53  44426.47  46114.93
103.9848_0.77 32147.61 28046.14  31520.55  31189.66  31365.98
113.0591_0.79 38483.06 44383.50  36327.01  45701.52  56593.10
115.063_0.79  10863.39 11022.03   8331.93  10238.20  12085.90
117.0788_0.79 91602.31 99711.00 172177.27 114091.65 112652.74
131.07_0.75   20844.59 18283.11  16641.86  13725.49  12609.90</code></pre>
</div>
</div>
</section>
</section>
</section>
<section id="preprocesamiento" class="level1">
<h1>Preprocesamiento</h1>
<p>Empezamos extrayendo la matriz de datos del contenedor.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>data_matrix <span class="ot">&lt;-</span> <span class="fu">assay</span>(se, <span class="st">"counts"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="filtrado" class="level2">
<h2 class="anchored" data-anchor-id="filtrado">Filtrado</h2>
<p>Si hay NAs eliminaremos las muestras que superen un porcentaje dado.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">anyNA</span>(data_matrix)) {</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">"Se han detectado valores NA. Aplicando filtrado e imputaci√≥n..."</span>)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filtrado: eliminar features con m√°s del 20% de NA</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  na_prop <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">is.na</span>(data_matrix))</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  data_matrix_filt <span class="ot">&lt;-</span> data_matrix[na_prop <span class="sc">&lt;=</span> <span class="fl">0.2</span>, ]</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Imputaci√≥n: reemplazar NA con el m√≠nimo valor positivo de cada muestra</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>  data_matrix_imp <span class="ot">&lt;-</span> <span class="fu">apply</span>(data_matrix_filt, <span class="dv">2</span>, <span class="cf">function</span>(x) {</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>    x[<span class="fu">is.na</span>(x)] <span class="ot">&lt;-</span> <span class="fu">min</span>(x[x <span class="sc">&gt;</span> <span class="dv">0</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>    x</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">"No se han detectado valores NA. Continuando sin filtrado ni imputaci√≥n."</span>)</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>  data_matrix_imp <span class="ot">&lt;-</span> data_matrix</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="normalizaci√≥n" class="level2">
<h2 class="anchored" data-anchor-id="normalizaci√≥n">Normalizaci√≥n</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalizar por suma total (TIC-like)</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="co"># data_matrix_norm &lt;- sweep(data_matrix_imp, 2,</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                           colSums(data_matrix_imp), </span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="co">#                           FUN = "/")</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Transformaci√≥n logar√≠tmica</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>log_data <span class="ot">&lt;-</span> <span class="fu">log2</span>(data_matrix_imp <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Escalado z-score por muestra (centra y escala cada perfil)</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>data_matrix_norm <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(log_data)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Los datos normalizados se a√±aden como un nuevo slot al contenedor</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># A√±adir como nuevo assay al objeto SummarizedExperiment</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(se, <span class="st">"data_matrix_norm"</span>) <span class="ot">&lt;-</span> data_matrix_norm</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Guardar tabla final</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(data_matrix_norm,<span class="fu">file.path</span>(results_dir,<span class="st">"results_filtered_normalized.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="exploraci√≥n-de-los-datos" class="level1">
<h1>Exploraci√≥n de los datos</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>(data_matrix_norm),</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">horizontal =</span> <span class="cn">TRUE</span>,</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">"Distribuci√≥n de intensidades normalizadas por metabolito"</span>,</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">xlab =</span> <span class="st">"Intensidad (log2, escalada)"</span>,</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">col =</span> <span class="st">"lightblue"</span>,</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">border =</span> <span class="st">"grey40"</span>,</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">outline =</span> <span class="cn">FALSE</span>,</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">las =</span> <span class="dv">1</span>,</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">cex.axis =</span> <span class="fl">0.6</span>,</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">cex.main =</span> <span class="fl">0.9</span>,</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">cex.lab =</span> <span class="fl">0.6</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<section id="an√°lisis-de-correlacion" class="level2">
<h2 class="anchored" data-anchor-id="an√°lisis-de-correlacion">An√°lisis de correlacion</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrplot)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Calcular matriz de correlaciones entre muestras</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>cor_muestra <span class="ot">&lt;-</span> <span class="fu">cor</span>(<span class="fu">assay</span>(se, <span class="st">"data_matrix_norm"</span>))</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(cor_muestra)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 110 110</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualizar con corrplot</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">corrplot</span>(cor_muestra,</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">method =</span> <span class="st">"color"</span>,</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">type =</span> <span class="st">"upper"</span>,</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">order =</span> <span class="st">"hclust"</span>,</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">tl.cex =</span> <span class="fl">0.6</span>,</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">col =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"white"</span>, <span class="st">"red"</span>))(<span class="dv">200</span>),</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">"Correlaci√≥n entre muestras"</span>,</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/figure-html/explore1-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="analisis-factoriales" class="level2">
<h2 class="anchored" data-anchor-id="analisis-factoriales">Analisis factoriales</h2>
<p>Empezamos con un PCA</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(FactoMineR)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Transponer datos: individuos = muestras, variables = metabolitos</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>pca_res <span class="ot">&lt;-</span> <span class="fu">PCA</span>(<span class="fu">t</span>(data_matrix_norm), <span class="at">graph =</span> <span class="cn">FALSE</span>)</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Scree plot: porcentaje de varianza explicada por cada componente</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_eig</span>(pca_res, <span class="at">addlabels =</span> <span class="cn">TRUE</span>, <span class="at">barfill =</span> <span class="st">"skyblue"</span>, <span class="at">barcolor =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Scree plot - varianza explicada"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/figure-html/explore2-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Gr√°fico PCA con muestras</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_pca_ind</span>(pca_res,</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">geom.ind =</span> <span class="st">"point"</span>,</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">pointshape =</span> <span class="dv">21</span>,</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">fill.ind =</span> <span class="st">"blue"</span>,</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">col.ind =</span> <span class="st">"black"</span>,</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">repel =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"PCA de muestras (FactoMineR)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/figure-html/explore2-2.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="mfa" class="level2">
<h2 class="anchored" data-anchor-id="mfa">MFA</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="do">## An√°lisis factorial m√∫ltiple (MFA)</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(FactoMineR)</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. Matriz de expresi√≥n</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>X <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">assay</span>(se, <span class="st">"data_matrix_norm"</span>))</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. Preparar data.frame y a√±adir variable cualitativa</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>X_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(X)</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>X_df<span class="sc">$</span>obesity_type <span class="ot">&lt;-</span> <span class="fu">colData</span>(se)<span class="sc">$</span>obesity_type</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. Preparar info de grupos</span></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>platforms <span class="ot">&lt;-</span> <span class="fu">rowData</span>(se)<span class="sc">$</span>platform</span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>vars_by_platform <span class="ot">&lt;-</span> <span class="fu">split</span>(<span class="fu">rownames</span>(se), platforms)</span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>X_df <span class="ot">&lt;-</span> X_df[, <span class="fu">c</span>(<span class="fu">unlist</span>(vars_by_platform), <span class="st">"obesity_type"</span>)]</span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a>group_sizes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">sapply</span>(vars_by_platform, length), <span class="dv">1</span>)</span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a>group_types <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"c"</span>, <span class="fu">length</span>(vars_by_platform)), <span class="st">"n"</span>)  <span class="co"># √∫ltima es cualitativa</span></span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a>group_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">names</span>(vars_by_platform), <span class="st">"obesity"</span>)</span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a><span class="co"># 4. Ejecutar MFA</span></span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a>res_mfa <span class="ot">&lt;-</span> <span class="fu">MFA</span>(</span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">base =</span> X_df,</span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">group =</span> group_sizes,</span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> group_types,</span>
<span id="cb23-27"><a href="#cb23-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">name.group =</span> group_names,</span>
<span id="cb23-28"><a href="#cb23-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">num.group.sup =</span> <span class="fu">length</span>(group_sizes),  <span class="co"># √∫ltima posici√≥n es suplementaria</span></span>
<span id="cb23-29"><a href="#cb23-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">graph =</span> <span class="cn">FALSE</span></span>
<span id="cb23-30"><a href="#cb23-30" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_screeplot</span>(res_mfa, <span class="at">addlabels =</span> <span class="cn">TRUE</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">50</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 5. Visualizar individuos (muestras)</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_mfa_ind</span>(</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  res_mfa,</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">habillage =</span> <span class="st">"obesity_type"</span>,</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">addEllipses =</span> <span class="cn">TRUE</span>,</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">"red"</span>, <span class="st">"blue"</span>),  <span class="co"># Colores personalizados</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">label =</span> <span class="st">"none"</span>,              <span class="co"># No mostrar etiquetas de muestras</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">repel =</span> <span class="cn">FALSE</span>,</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">title =</span> <span class="st">"MFA - Muestras coloreadas por tipo de obesidad"</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extraer informaci√≥n de variables y grupos</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>mfa_vars <span class="ot">&lt;-</span> <span class="fu">get_mfa_var</span>(res_mfa)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualizar contribuci√≥n de grupos a los ejes (autom√°ticamente)</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_mfa_var</span>(</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  res_mfa,</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"group"</span>,</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> <span class="st">"Dark2"</span>,</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">title =</span> <span class="st">"Contribuci√≥n de los grupos a los componentes principales"</span></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<section id="matriz-de-factores" class="level3">
<h3 class="anchored" data-anchor-id="matriz-de-factores">Matriz de factores</h3>
<p>El an√°lisis factorial (m√∫ltiple) genera una nueva matriz con las mismas filas y los factores o variables latentes como columnas.</p>
<p>Esta nueva matriz puede utilizarse de forma impl√≠cita o expl√≠cita para an√°lisis posteriores como la b√∫squeda de metabotipos.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>factor_coords <span class="ot">&lt;-</span> res_mfa<span class="sc">$</span>ind<span class="sc">$</span>coord</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Ver dimensiones</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(factor_coords)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 110   5</code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualizar primeras filas</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(factor_coords)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           Dim.1      Dim.2      Dim.3       Dim.4      Dim.5
C1030 -0.6321534 -0.2029206 -0.2409442 -0.19341785 -0.9538281
C1131 -2.7197222  1.2718608 -0.3780766 -1.19539167  0.8745664
C1133  0.3053819 -0.9254186  0.1501404  0.06059781 -0.2166434
C1171 -0.2994113 -0.1788080  0.1994838 -0.54755132 -0.6400878
C1204 -1.2388472 -0.5104731  0.9027184 -1.19008276  0.2045216
C1222  2.3403716  1.0010518  1.5032099 -0.06406640 -0.3264256</code></pre>
</div>
</div>
</section>
</section>
<section id="clustering-y-obesidad" class="level2">
<h2 class="anchored" data-anchor-id="clustering-y-obesidad">Clustering y obesidad</h2>
<section id="clustering-y-relaci√≥n-con-la-obesidad" class="level3">
<h3 class="anchored" data-anchor-id="clustering-y-relaci√≥n-con-la-obesidad">üîç Clustering y relaci√≥n con la obesidad</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 11. Clustering jer√°rquico sobre los factores</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>res_hcpc <span class="ot">&lt;-</span> <span class="fu">HCPC</span>(res_mfa, <span class="at">graph =</span> <span class="cn">FALSE</span>)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 12. Visualizaci√≥n de cl√∫steres</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_cluster</span>(</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>  res_hcpc,</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">geom =</span> <span class="st">"point"</span>,</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> <span class="st">"jco"</span>,</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">repel =</span> <span class="cn">TRUE</span>,</span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">show.clust.cent =</span> <span class="cn">TRUE</span>,</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">"Clustering de muestras basado en MFA"</span></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Rean√°lisis_de_resultados_procesados-Estudio_ST002993_files/figure-html/hcpc-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 13. Comparar cl√∫ster con tipo de obesidad</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>cluster_assignments <span class="ot">&lt;-</span> res_hcpc<span class="sc">$</span>data.clust<span class="sc">$</span>clust</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>obesity_type <span class="ot">&lt;-</span> <span class="fu">colData</span>(se)<span class="sc">$</span>obesity_type</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="at">Cluster =</span> cluster_assignments, <span class="at">Obesity =</span> obesity_type)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       Obesity
Cluster Idiopathic obesity Monogenic obesity
      1                  0                 1
      2                 14                17
      3                 41                37</code></pre>
</div>
</div>
</section>
</section>
</section>
<section id="almacenaje-de-datos-y-resultados" class="level1">
<h1>Almacenaje de datos y resultados</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(se, res_mfa, factor_coords,  res_hcpc, <span class="at">file =</span><span class="fu">file.path</span>(results_dir, <span class="st">"ST002993_processed_objects.RData"</span>))</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Guardar tabla final</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(factor_coords,<span class="fu">file.path</span>(results_dir,<span class="st">"factor_matrix.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>


<div id="quarto-appendix" class="default"><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>

<ol>
<li id="fn1"><p>L‚Äôestudi original fa servir AF. Aqui fare servir MFA que sembla casi mes raonable al haver plataformes diferents<a href="#fnref1" class="footnote-back" role="doc-backlink">‚Ü©Ô∏é</a></p></li>
<li id="fn2"><p>Al repositori nomes hi ha les dades de quatre de les plataformes. La cinquena s‚Äôomet de l‚Äôan√†lisi<a href="#fnref2" class="footnote-back" role="doc-backlink">‚Ü©Ô∏é</a></p></li>
</ol>
</section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "Óßã";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>